sdk/state/state.go

```diff
@@ -45,12 +45,15 @@ type Snapshot struct {
 	RemoteEscrowSequence       int64
 	RemoteEscrowAccountBalance int64
 
-	OpenAgreement            OpenAgreement
-	OpenExecutedAndValidated bool
-	OpenExecutedWithError    bool
-
-	LatestAuthorizedCloseAgreement   CloseAgreement
-	LatestUnauthorizedCloseAgreement CloseAgreement
+	OpenAgreement             OpenAgreement
+	OpenAgreementTransactions OpenAgreementTransactions
+	OpenExecutedAndValidated  bool
+	OpenExecutedWithError     bool
+
+	LatestAuthorizedCloseAgreement               CloseAgreement
```

Comment:
❓ in the [ticket](https://github.com/stellar/experimental-payment-channels/issues/199) you used the idea of a packaged field that grouped the agreements with their transactions. I initially liked that to decrease the chance of the transactions getting out of sync with the agreement
eg. updating `channel.latestAuthorizedCloseAgreementPackage` at one time vs updating `channel. latestAuthorizedCloseAgreement` and then `channel.latestAuthorizedCloseAgreementTransactions` afterward

I'm not pushing for the single field as it doesn't _really_ protect them getting out of sync, but just wondering what your thought process was for not packaging them.

Reply:
💯 I really like that idea, but for a couple reasons I deferred doing that. CloseAgreement is like this unit of data that needs to be communicated with the other participant. Everything in it needs communicating. If we bundle transactions in there, then there is peripheral information in there. So, if we are going to do thed single object I'd probably want to create a new object that wraps the CloseAgreement and the CloseAgreementTransactions. But doing so would require touching every line of code that touches the existing fields. That's a big refactor which I think we should do independently of adding a new feature (this PR). I'm up for following up on that to at least see what it would look like.

Reply:
sgtm for following up with. I'm not a huge advocate for it, only thing I was thinking is it would be one way to help ensure the agreements and transactions stay in sync since separately stored now. There also might be other/better ways that become clearer later, or if it's even a problem at all

Reply:
💯 I think it is compelling for that reason too.